components/script/script_runtime.rs

```diff
@@ -283,7 +283,8 @@ unsafe extern "C" fn enqueue_promise_job(
     result
 }
 
-#[allow(unsafe_code, crown::unrooted_must_root)]
+#[allow(unsafe_code)]
+#[cfg_attr(feature = "crown", allow(crown::unrooted_must_root))]
```

Comment:
We already pass `--cfg=crown` when building with `crown` so could this just be `#[cfg_attr(crown, allow(crown::unrooted_must_root))]`? If not, I think we'll need to make sure that the `crown` feature is enabled in `run_cargo_like_build_command` in `command_base.py`.

Reply:
I don't know if we can enable features in dependent crates (mozjs) based on a cfg attribute.

Reply:
Hrm. it is a bit odd that it is `cfg(crown)` in Servo but a feature in `mozjs`. Would it make more sense to use `cfg` everywhere or a feature everywhere?

Reply:
It's certainly possible to switch from a feature to a cfg in mozjs. I don't have a good sense of which is better though!

Reply:
It does seem like `cfg(crown)` makes more sense as we are turning it on for the entire workspace. Maybe we can use a feature here and convert mozjs to use `cfg` in a followup? I think the only thing missing is to enable the feature with `mach` here.

Reply:
The idea was that crown stuff would be behind cfg eventually: https://github.com/servo/servo/pull/32494#issuecomment-2181504756 as we want it to apply to whole workspace from outside (if crown is used).

Reply:
Yeah, I'll go ahead with converting script to use the cfg and turn on the feature so it propagates to mozjs, and convert mozjs in a followup.